src/Common/Dlgcode.c

```diff
@@ -9898,9 +9897,12 @@ HANDLE DismountDrive (wchar_t *devName, wchar_t *devicePath)
 	}
 
 	if (!bResult)
```

Comment:
The both variants actually are incorrect. You have to use __try/__finally to invoke the cleanup code because of structured exceptions in the WinAPI.